### PR TITLE
fix: use orderField and createdAfter to unify for other plugins

### PR DESCRIPTION
**Pull request by mareklibra**
## Hey, I just made a Pull Request!

In the Notifications plugin,
- the sort/sortOrder URL query params are replaced by `orderField` to unify with the Catalog
- the `created_after` param is camelCased to `createdAfter`

Follow-up for: https://github.com/backstage/backstage/pull/23351

#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [x] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

**Comment by mareklibra** [comment]
Rebased

File: plugins/catalog-backend/api-report.md [hunk 1]
```diff
@@ -389,6 +395,11 @@ export const locationSpecToLocationEntity: typeof locationSpecToLocationEntity_2
 // @public @deprecated (undocumented)
 export const locationSpecToMetadataName: typeof locationSpecToMetadataName_2;
 
+// @public (undocumented)
+export function parseEntityOrderFieldParams(
```
Comment (Rugvip) [inline plugins/catalog-backend/api-report.md:399]: Let's skip this and just copy it for now. We don't want the notifications-backend to depend on catalog-backend anyway. If this becomes a frequent thing we can look for an opportunity to provide some common utilities as part of the framework instead

File: .changeset/poor-coats-boil.md [hunk 2]
```diff
@@ -0,0 +1,6 @@
+---
+'@backstage/plugin-notifications-backend': minor
+'@backstage/plugin-notifications': minor
+---
+
+notifications-backend URL query parameters changed from sort/sortOrder to orderField and created_after to createdAfter to unify with other plugins
```
Comment (drodil) [inline .changeset/poor-coats-boil.md:6]: Capitalize this, check the Vale.spelling errors

**Review by Rugvip — APPROVED** [review]
Sweet! 🎉 

Let's :shipit:

**Pull request merged**